src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -120,6 +128,54 @@ public override void VisitParenthesizedLambdaExpression(ParenthesizedLambdaExpre
             }
         }
 
+        public override void VisitLocalFunctionStatement(LocalFunctionStatementSyntax node)
+        {
+            var body = (CSharpSyntaxNode)node.Body ?? node.ExpressionBody;
+            MethodSymbol match = null;
+            // Don't use LookupLocalFunction because it recurses up the tree, as it
+            // should be defined in the directly enclosing block (see note below)
+            foreach (var candidate in _enclosing.LocalFunctions)
+            {
+                if (candidate.Name == node.Identifier.Text)
```

Comment:
You probably want to match by location, both to support a possible extension to support overloading and to support error scenarios where a name is multiply defined.